### PR TITLE
Feature/juicy damage indicators

### PR DESCRIPTION
**Pull request by mariofv**
What's new:
- `DamageIndicator` trajectory is parabloic.
- `DamageIndicator` trajectory is determined by hit direction.
- `DamageIndicator` font is bolder.
- `DamageIndicator` color changes with power of the hit.
- `DamageIndicator` size changes with power of the hit.

### CODE REVIEW

**Review by JorxSS — APPROVED** [review]
This is pretty cool! GJ!!

image

File: Scripts/src/Script/DamageIndicator.cpp [hunk 1]
```diff
@@ -41,36 +43,82 @@ void DamageIndicator::Start()
...
 }
 
 bool DamageIndicator::IsAlive() const
 {
-	return alive;
+	return state != DamageIndicatorState::DEAD;
 }
 
-void DamageIndicator::Spawn(int number, float2 position)
+void DamageIndicator::Spawn(int number, float power, float2 position, const float2& direction)
```
Comment (Dartemiss) [inline Scripts/src/Script/DamageIndicator.cpp:102]: position could be const ref?

File: Scripts/src/Script/DamageIndicatorSpawner.cpp [hunk 2]
```diff
@@ -14,9 +16,12 @@
 
 
 #include "DamageIndicator.h"
+#include "PlayerAttack.h"
 
 #include "imgui.h"
 
+const float MAX_DAMAGE = 55.f;
```
Comment (Dartemiss) [inline Scripts/src/Script/DamageIndicatorSpawner.cpp:23]: Max damage isn't 55

Comment (JorxSS) [reply Scripts/src/Script/DamageIndicatorSpawner.cpp:23]: It is, and I told him that value, but now that I'm thinking, if we are going to do power ups, that maximum damage will increase

**Review by Dartemiss — APPROVED** [review]
Very good job man! I love the result. With a good sound effect depending on attack/damage too will be amazing.

File: Scripts/src/Script/DamageIndicatorSpawner.cpp (hunk 2 above)
Comment (Dartemiss) [reply Scripts/src/Script/DamageIndicatorSpawner.cpp:23]: Lightball can deal until 71 damage, so it isn't

File: Scripts/src/Script/DamageIndicator.cpp (hunk 1 above)
Comment (mariofv) [reply Scripts/src/Script/DamageIndicator.cpp:102]: No, because I'm modifying position later in the method

**Pull request merged**